managed/services/telemetry/telemetry.go

```diff
@@ -225,6 +228,22 @@ func (s *Service) prepareReport(ctx context.Context) *pmmv1.ServerMetric {
 	}
 
 	for _, telemetry := range s.config.telemetry {
+		if telemetry.Extension != "" {
+			extension, ok := s.extensions[telemetry.Extension]
+			if ok {
+				metrics, err := extension.FetchMetrics(ctx, telemetry)
+				if err != nil {
+					s.l.Debugf("failed while calling extension [%s]:%s", telemetry.Extension, err)
+					continue
+				}
+				telemetryMetric.Metrics = append(telemetryMetric.Metrics, metrics...)
+			} else {
+				s.l.Errorf("telemetry extension [%s] is not supported", telemetry.Extension)
+			}
```

Comment:
```suggestion
			if !ok {
				s.l.Errorf("telemetry extension [%s] is not supported", telemetry.Extension)
				continue
			}

			metrics, err := extension.FetchMetrics(ctx, telemetry)
			if err != nil {
				s.l.Debugf("failed while calling extension [%s]:%s", telemetry.Extension, err)
			}
			telemetryMetric.Metrics = append(telemetryMetric.Metrics, metrics...)

```

Reply:
Its little bit less nested, but its up to you.

Reply:
but 1 more continue, and inderection. IMO it is personal style preference.

Reply:
discussed offline, code is updated